EIPS/eip-5601.md

```diff
@@ -0,0 +1,2123 @@
+
+### Extendable
+
+The core Extendable contract keeps track of its Extensions through a set of mappings.
+
+![Extendable mapping Extensions](../assets/eip-xxxx/mapping.png)
+
+#### Extending
+
+When it is extended with an Extension, it records the Extension address against the interface that the Extension implements as a key-value pair. The extended contract now is marked to implement the interface implemented by the Extension. By extending more Extensions, the Extendable contract can have an evolving and growing interface as a collection of all Extensions that it has.
+
+![Extending](../assets/eip-xxxx/extend.png)
+
+#### Calling functions
+
+When a function is called, the Extendable contract identifies which Extension contains the desired functionality and performs a delegatecall.
+
+![Delegating](../assets/eip-xxxx/delegate.png)
+
+Delegatecalls allow one contract to execute the function of another contract as if the original contract itself has the function. It shifts the context of the execution of the function to that of the delegator, but using the function logic of the delegatee.
+
+![Delegating call flow](../assets/eip-xxxx/delegation.png)
+
+#### Accessing state variables
+
+Accessing storage variables are done using slot assignment. Each Extension that has functions that use storage variables imports the storage that they intend to interact with as a "storage module" library. Such a library defines the variables and exposes a function that returns them. When defining a library, the assigned slot address where the variables exist must be chosen with strict uniqueness in mind, else risks the security of the contract. An Extendable contract may have several different storage modules and each one will exist at its assigned slot.
+
+![Storage slot address](../assets/eip-xxxx/addressing.png)
+
+Each Extendable contract owns its own addressable storage space, and Extensions, when called, will access the current context's contract when interacting with contract state. This allows Extensions to share storage module definitions as the variables themselves will indexed against each Extendable contract instance during execution.
+
+![Storage slot access](../assets/eip-xxxx/access.png)
+
+---
+
+As a reminder, this document serves to define a set of conceptual primitives and their associated building blocks to allow developers to follow an approach for writing extendable smart contracts. The reference implementation can be found [here](https://github.com/violetprotocol/extendable/).
+
+## Specification
+
+The core specification contains the main primitive generalisations of the Extendable pattern's building blocks. Any reference implementations or utilities that enhance the framework for developers is included in the [appendix](#appendix).
+
+### Terminology
+
+Extendable - Refers to the
+Extension - 
+Storage Library - 
+Extend - 
+Retract - 
+Replace - 
+Delegatecall -
+Function Selector - 
+
+### ERC165 Singleton
```

Comment:
If you're referring to the interface, you can put the interface name in backticks \`ERC165\`, otherwise you'll need to link to the EIP here.